### PR TITLE
initiate internLM2_chat_1.8b_llava tutorial

### PR DESCRIPTION
**Pull request by maxchiron**
- [x] create read.me tutorial file. (not finished.)
- [x] create fake data for finetune.
- [x] add test image.

### CODE REVIEW

File: xtuner/llava/xtuner_llava.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+    ]
+  }
+]
+```
+The questions and answers, please generate for me, based on the image I sent to you. Thes questions should be from the shallow to the deep, and the answers should be as detailed and correct as possible. The questions and answers should be stick to the contents in the image itself, like objects, peoples, equipment, environment, purpose, color, attitude, etc. 5 question and answer pairs.
+</details>
+<br>
+
+为了方便大家跟随课程，针对这张示例图片的问答对数据（repeat_data.json），大家按照下面的脚本运行就可以生成啦~（重复10000次，生成的文件一百多MB）
+
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:285]: 请添加详细的环境配置教程，包括conda环境配置，和 LLM 部分解耦~

File: xtuner/llava/xtuner_llava.md [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+```bash
+
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:363]: 此处命令应该新增 cd xxx 路径的命令， 用户可能不知道在哪个路径下执行那个命令

File: xtuner/llava/xtuner_llava.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+git clone https://github.com/InternLM/Tutorial -b camp2 && cd tutorial/xtuner/llava && conda activate xtuner0.1.17
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:364]: Tutorial 的 T和 t 大小需要统一

File: xtuner/llava/iter_2181.pth
Comment (vansin) [inline xtuner/llava/iter_2181.pth:1]: 21 MB 的权重文件放在 GitHub 中太大了，放在 share 路径下比较好

File: xtuner/llava/xtuner_llava.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+- data_path = data_root + 'LLaVA-Instruct-150K/llava_v1_5_mix665k.json'
++ data_path = data_root + 'repeated_data.json'
+- image_folder = data_root + 'llava_images'
++ image_folder = data_root
+```
+
+#### 1.3.4.3. 开始Finetune
+
+```bash
+xtuner train /root/tutorial/xtuner/llava_data/llava_internlm2_chat_1_8b_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune_copy.py --deepspeed deepspeed_zero2
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:418]: llava_data 是不是写错了，llava

File: xtuner/llava/xtuner_llava.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+export MKL_THREADING_LAYER=GNU
+
+# pth转huggingface
+xtuner convert pth_to_hf llava_internlm2_chat_1_8b_clip_vit_large_p14_336_e1_gpu8_pretrain /root/tutorial/xtuner/llava_data/iter_2181.pth /root/tutorial/xtuner/llava_data/iter_2181_hf
+
+# 启动！
+xtuner chat internlm/internlm2-chat-1_8b --visual-encoder openai/clip-vit-large-patch14-336 --llava /root/tutorial/xtuner/llava_data/iter_2181_hf --prompt-template internlm2_chat --image /root/tutorial/xtuner/llava_data/llava_data/test_img/oph.jpg
+```
+
+#### 1.3.5.2. Finetune后
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:438]: 两个问题的英文直接贴在文档中方便用户直接 copy ，不需要用户自己敲

File: xtuner/llava/xtuner_llava.md [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+]
+```
+The questions and answers, please generate for me, based on the image I sent to you. Thes questions should be from the shallow to the deep, and the answers should be as detailed and correct as possible. The questions and answers should be stick to the contents in the image itself, like objects, peoples, equipment, environment, purpose, color, attitude, etc. 5 question and answer pairs.
+</details>
+<br>
+
+为了方便大家跟随课程，针对这张示例图片的问答对数据（repeat_data.json），大家按照下面的脚本运行就可以生成啦~（重复10000次，生成的文件一百多MB）
+
+```bash
+git clone https://github.com/InternLM/Tutorial -b camp2 && cd tutorial/xtuner/llava && conda activate xtuner0.1.17
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:363]: 在 git clone 前，需要线 cd 到某个路径，用户很容易在不同的路径下执行，导致后面的操作全部崩

File: xtuner/llava/xtuner_llava.md [hunk 7]
```diff
@@ -0,0 +1,479 @@
...
+#### 1.3.4.2. 准备配置文件
+
+##### 1.3.4.2.1. 创建配置文件
+
+```bash
+# 查询xtuner内置配置文件
+xtuner list-cfg -p llava_internlm2_chat_1_8b
+
+# 拷贝配置文件到当前目录（注意命令最后有个英文句号
+xtuner copy-cfg llava_internlm2_chat_1_8b_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune .
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:380]: xtuner copy-cfg llava_internlm2_chat_1_8b_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune .

中的 .  需要替换为绝对路径，然后命令行整体换行

File: xtuner/llava/xtuner_llava.md [hunk 8]
```diff
@@ -0,0 +1,479 @@
...
+find /root/tutorial/xtuner/llava_data -type d -name "iter_3750.pth"
+
+# pth转huggingface
+xtuner convert pth_to_hf "$(find /root/tutorial/xtuner/llava_data -type d -name "iter_3750.pth")" /root/tutorial/xtuner/llava_data/iter_3750_hf
+
+
+# 启动！
+xtuner chat internlm/internlm2-chat-1_8b \
+  --visual-encoder openai/clip-vit-large-patch14-336 \
+  --llava /root/tutorial/xtuner/llava_data/iter_3750_hf \
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:463]: 此处 llava_data 的路径是不是也有问题

File: xtuner/llava/xtuner_llava.md [hunk 9]
```diff
@@ -0,0 +1,479 @@
...
+```bash
+# 解决小bug
+export MKL_SERVICE_FORCE_INTEL=1
+export MKL_THREADING_LAYER=GNU
+
+# 找到finetune结束的pth文件位置
+find /root/tutorial/xtuner/llava_data -type d -name "iter_3750.pth"
+
+# pth转huggingface
+xtuner convert pth_to_hf "$(find /root/tutorial/xtuner/llava_data -type d -name "iter_3750.pth")" /root/tutorial/xtuner/llava_data/iter_3750_hf
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:457]: 为什么需要用 find 命令，感觉可以直接用 绝对路径的命令

File: xtuner/llava/xtuner_llava.md [hunk 10]
```diff
@@ -0,0 +1,479 @@
...
+      }
+    ]
+  }
+]
+```
+The questions and answers, please generate for me, based on the image I sent to you. Thes questions should be from the shallow to the deep, and the answers should be as detailed and correct as possible. The questions and answers should be stick to the contents in the image itself, like objects, peoples, equipment, environment, purpose, color, attitude, etc. 5 question and answer pairs.
+</details>
+<br>
+
+为了方便大家跟随课程，针对这张示例图片的问答对数据（repeat_data.json），大家按照下面的脚本运行就可以生成啦~（重复10000次，生成的文件一百多MB）
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:360]: 重复 200 次就够了，batch_size 改为1，然后卡选 30% A100

File: xtuner/llava/xtuner_llava.md [hunk 11]
```diff
@@ -0,0 +1,479 @@
...
+
+# Data
+- data_root = './data/llava_data/'
++ data_root = './llava_data/'
+- data_path = data_root + 'LLaVA-Instruct-150K/llava_v1_5_mix665k.json'
++ data_path = data_root + 'repeated_data.json'
+- image_folder = data_root + 'llava_images'
++ image_folder = data_root
+```
+
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:417]: # Model
llm_name_or_path = 'internlm/internlm2-chat-1_8b'
visual_encoder_name_or_path = 'openai/clip-vit-large-patch14-336'
# Specify the pretrained pth
pretrained_pth = './iter_2181.pth'


此处三个都改成绝对路径，有一些 share 路径里面能够直接取到

File: xtuner/llava/xtuner_llava.md [hunk 12]
```diff
@@ -0,0 +1,497 @@
...
+- data_path = data_root + 'LLaVA-Instruct-150K/llava_v1_5_mix665k.json'
++ data_path = data_root + 'repeated_data.json'
+- image_folder = data_root + 'llava_images'
++ image_folder = data_root
+
+# Scheduler & Optimizer
+- batch_size = 16  # per_device
++ batch_size = 1  # per_device
+```
+
```
Comment (vansin) [inline xtuner/llava/xtuner_llava.md:430]: 提供一个完整已经改好的配置文件供用户 copy
